backend/fms_core/models/sample.py

```diff
@@ -279,23 +281,37 @@ def add_error(field: str, error: str):
             )
 
         if self.extracted_from:
-            if self.extracted_from.sample_kind.name in Sample.BIOSPECIMEN_TYPES_NA:
+            try:
+                ProcessSample = apps.get_model("fms_core", "ProcessSample")
+
+                sample_lineage = SampleLineage.objects.get(child_id=self.id)
```

Comment:
Not sure why you insist on using this approach instead of the relation (child_of) if it forces you to use exception handling to get it to work...

Reply:
noted. If I can suggest using a different name than 'child_of' for the attribute, I had forgotten it was a SampleLineage, I thought child_of meant the parent Sample.

Reply:
child_of returns the parent sample but can be filtered with any value of the intermediate table, including ProcessSample. This is up to you i really just dislike exception handling of normal operations.